### ISSUE TITLE
Colons in path segments are not interpreted correctly

### ISSUE THREAD

**Issue by dmurvihill** [lemonlabsuk/scala-uri#192]
**Describe the bug**
Tested on Scala 2.11, scala-uri 1.4.10 (let me know if this has already been fixed in 2.0)

**To Reproduce**
```scala
val url = n+Pw+yb-k.+czhe+p.d.kgx-.s-8++..t+-.6-.s+-+-gi://32.60.46.58:3469/@@%f2@%4e::n@=:@:.:v*$%43*:%65@@:@/:@%89%c3,z@%4e;@c~+%08@_':'::%B2%e0@$@(o%84J%7f+::%81::%25@%2be-%ED+_~::%62.%62%23%BF/:'+:::@=i::=-%A0:%05@:_::@@!@;%fb):g:+@!*7%1B@:)@)x@c%34:)%f2@:%36)/%F1%5b:.%cd!::%35&@@%eaw.::m%9B:,%Ca:e%44::(@:/+@:@@:@@*.-(+%69r'%C1@n-0::E:/*%36%67,%4f'%A3%f9@@%27:=@@!@6t%30@:*_~@$@@%df():.@:x'_%02:@~@!::%63%7B%B4@%d0%39-_%e8@:/@'D:@-%56@%cF@%f7v:%c7(!;~)@_:y:d*::_@+%6d%E8@%1D:&@@.(:%DE:@=g(@://@@%a6@x@.&&@-%80':-::@_b%b9~%c9:%d1)@&:@,$@(*%0Ef@@~):%63'%30:!@@!:q:%4A%44%c8/)7-%42)g%a9::%A1/5%2b%cb'%aF::@/:L%1D:_%A4:@%76@::@%64@:/==Nt,&@-:@w~@,!,:%7Bm@i@@%01:@@:@@$:%b5@:@@&+f:@@:=@::@@:-%0c@::/%E5h%e4@3%61).@:%67%4E:@y%c2$@@%85:@r@@=W%bE%b5%1C@*%3e%f4%10_:%B1_@::+:%20::@_@(:(/:=~:::=~,++%25@:$:h~%3C@%DD_:%4es@:@@@.-:%c1%5D@@()~.-:%C2:n%79:&/@@%C9@)/e@@@:@%9D!::!@@@&$@%BA/:@$:(,@:@_$_%12:&.:@%63@&,~%d7%08:-@y@$@@%53(:x%9E;@::%6E:~%9E%88(;*o%71/w:@%aa-%D0%DA.%5d~%83:=':%2f:g@%34@%DC;@@%d7@h~_:,:::@@:%f7%73:~:c%42~:s%65:&l@+_/@%88:%48@:@::,/!/@:z,-@%3b:@%82*@@&%63%E5~:+,::,@@%9eg@;!'@::%DC@%44'/:w::%5ea%F9~%31$:*%0fc,@@:':%c9_:(=%fF2::%02-$b%42M/:)x8::,@~_@+U:-_/%Fa@)_@),%D1):@:,='@:_1@;:@+%33*:%cB/%98%bA)'@@),%cF(.$@&%44)%d7'!::%9A_:*-%9A:%89!$,:;~@%E0%9f$$;:*!:l%13@@-*c!:u:,/=@/%0E%6C::%EF%0A+s@(@%EEwi@@:%36%e93u/.q~J.(&%C9%ec_@@~%21.+@(-%59%7dK;@%ee@_:/%AC1.:*$(.:@'@*:&%80.::@@:+%86@%33@*%61::$_:M%53@@@)=%3e!+':;$o/@1$~@@r%E4_z::&%78:@%e0@v@:%66@@/:i*:~~%49~!%1E@:_@.&%Cc:%78:e/_@_%43%aC$l@x@x::;~(@=%01:r%2F:7@:o:@@@j@-@%eC%08!6:~@=!_~/@%83@%86:::%19/@%cC%93-@3/:_@'::@%Fek:%36@:(!;:%71@&:@@(%e2::/:@:/+&m@5@&@$::@%C3%2d@.:)v@+m~:::-%1F@yI_@%48n@%5a::%43@;@p/*@.%72--::='_:%4A~:l(:@.:%3b,g!:~:$-_;@+=%75k%e5:@;l*@@g/m:&:@,_::z-:6~:%F9a=/:-%e3%Ec%80@@)@='@.+x~&%a9%e9'%35@+-:+z@/%c8:%5d%52@%ca_~x@@:a%41!@./$:w,,(%3f@/'%bE@_%99@%b2/=mrd%3dy%3a:/:%85%30:@(%a0%FF@:9=%1c::@,%5F@q:~@@/%c7:$%86c/%10/_%e4-,::)%a8%ab:(__:@:%B3@k@(%9Da:@:@.u:@$&x@-:%1eh)::%94@%9b%34@:/_%3e:%0B%65%d0%1dC:@d@@@%bbe.%CC:p@::@@@%96::%8a.%8e$@%2B%C3:Z;.;.@@:@i!.+@@z$=:/@%17:@+:~%aF=%d8&@O@@@:-::@::o/@:.@.::'%63!_%F6m:%c4?//s//?%EF/?@?!@/:?~?*.?/:?//?&%48////?,?//??@/#%80?//:/,.???y//:=?/:@//:??:;/?*????/@&/@@k?/%b5?*?@

Url.parse(url)
```

```
[info]     Message: Invalid URL could not be parsed. Invalid input ':', expected _int (line 1, column 77):
[info]   n+Pw+yb-k.+czhe+p.d.kgx-.s-8++..t+-.6-.s+-+-gi://32.60.46.58:3469/@@%f2@%4e::n@=:@:.:v*$%43*:%65@@:@/:@%89%c3,z@%4e;@c~+%08@_':'::%B2%e0@$@(o%84J%7f+::%81::%25@%2be-%ED+_~::%62.%62%23%BF/:'+:::@=i::=-%A0:%05@:_::@@!@;%fb):g:+@!*7%1B@:)@)x@c%34:)%f2@:%36)/%F1%5b:.%cd!::%35&@@%eaw.::m%9B:,%Ca:e%44::(@:/+@:@@:@@*.-(+%69r'%C1@n-0::E:/*%36%67,%4f'%A3%f9@@%27:=@@!@6t%30@:*_~@$@@%df():.@:x'_%02:@~@!::%63%7B%B4@%d0%39-_%e8@:/@'D:@-%56@%cF@%f7v:%c7(!;~)@_:y:d*::_@+%6d%E8@%1D:&@@.(:%DE:@=g(@://@@%a6@x@.&&@-%80':-::@_b%b9~%c9:%d1)@&:@,$@(*%0Ef@@~):%63'%30:!@@!:q:%4A%44%c8/)7-%42)g%a9::%A1/5%2b%cb'%aF::@/:L%1D:_%A4:@%76@::@%64@:/==Nt,&@-:@w~@,!,:%7Bm@i@@%01:@@:@@$:%b5@:@@&+f:@@:=@::@@:-%0c@::/%E5h%e4@3%61).@:%67%4E:@y%c2$@@%85:@r@@=W%bE%b5%1C@*%3e%f4%10_:%B1_@::+:%20::@_@(:(/:=~:::=~,++%25@:$:h~%3C@%DD_:%4es@:@@@.-:%c1%5D@@()~.-:%C2:n%79:&/@@%C9@)/e@@@:@%9D!::!@@@&$@%BA/:@$:(,@:@_$_%12:&.:@%63@&,~%d7%08:-@y@$@@%53(:x%9E;@::%6E:~%9E%88(;*o%71/w:@%aa-%D0%DA.%5d~%83:=':%2f:g@%34@%DC;@@%d7@h~_:,:::@@:%f7%73:~:c%42~:s%65:&l@+_/@%88:%48@:@::,/!/@:z,-@%3b:@%82*@@&%63%E5~:+,::,@@%9eg@;!'@::%DC@%44'/:w::%5ea%F9~%31$:*%0fc,@@:':%c9_:(=%fF2::%02-$b%42M/:)x8::,@~_@+U:-_/%Fa@)_@),%D1):@:,='@:_1@;:@+%33*:%cB/%98%bA)'@@),%cF(.$@&%44)%d7'!::%9A_:*-%9A:%89!$,:;~@%E0%9f$$;:*!:l%13@@-*c!:u:,/=@/%0E%6C::%EF%0A+s@(@%EEwi@@:%36%e93u/.q~J.(&%C9%ec_@@~%21.+@(-%59%7dK;@%ee@_:/%AC1.:*$(.:@'@*:&%80.::@@:+%86@%33@*%61::$_:M%53@@@)=%3e!+':;$o/@1$~@@r%E4_z::&%78:@%e0@v@:%66@@/:i*:~~%49~!%1E@:_@.&%Cc:%78:e/_@_%43%aC$l@x@x::;~(@=%01:r%2F:7@:o:@@@j@-@%eC%08!6:~@=!_~/@%83@%86:::%19/@%cC%93-@3/:_@'::@%Fek:%36@:(!;:%71@&:@@(%e2::/:@:/+&m@5@&@$::@%C3%2d@.:)v@+m~:::-%1F@yI_@%48n@%5a::%43@;@p/*@.%72--::='_:%4A~:l(:@.:%3b,g!:~:$-_;@+=%75k%e5:@;l*@@g/m:&:@,_::z-:6~:%F9a=/:-%e3%Ec%80@@)@='@.+x~&%a9%e9'%35@+-:+z@/%c8:%5d%52@%ca_~x@@:a%41!@./$:w,,(%3f@/'%bE@_%99@%b2/=mrd%3dy%3a:/:%85%30:@(%a0%FF@:9=%1c::@,%5F@q:~@@/%c7:$%86c/%10/_%e4-,::)%a8%ab:(__:@:%B3@k@(%9Da:@:@.u:@$&x@-:%1eh)::%94@%9b%34@:/_%3e:%0B%65%d0%1dC:@d@@@%bbe.%CC:p@::@@@%96::%8a.%8e$@%2B%C3:Z;.;.@@:@i!.+@@z$=:/@%17:@+:~%aF=%d8&@O@@@:-::@::o/@:.@.::'%63!_%F6m:%c4?//s//?%EF/?@?!@/:?~?*.?/:?//?&%48////?,?//??@/#%80?//:/,.???y//:=?/:@//:??:;/?*????/@&/@@k?/%b5?*?@
```

Excuse the confusing URL, it was auto-generated by ScalaCheck and the shrinking is not working correctly.

The error occurs on the first occurrence of `::`, when trying to parse the second colon; it seems to think it is parsing a port number even though there is a port number earlier in the URL.

It may think it is parsing a `path-noscheme` which would not allow colon characters; however, this URL has the scheme `n+Pw+yb-k.+czhe+p.d.kgx-.s-8++..t+-.6-.s+-+-gi`, which is valid. So it should be parsing a `path-abempty` which would interpret a colon as just another character in the path segment.

**Comment by theon** [comment]
Thanks for raising this. The bug does also exist in the most current version of scala-uri.

The issue was that the parser was trying to read the URL in the form scheme://user:password@host:port where in your example, password was `3469/` rather than interpreting the `/` to be the end of the host+port.

**Comment by theon** [comment]
This has been released in version `2.2.5`
